clear-intermediate-datasets/python-runnables/clear_int_datasets/runnable.py

```diff
@@ -66,13 +66,15 @@ def run(self, progress_callback):
             rcp_outputs_dict = rcp["outputs"]
             # CASE: no input dataset
             if rcp_inputs_dict:
-                input_key = list(rcp_inputs_dict.keys())[0]
-                rcp_inputs_list = [x["ref"] for x in rcp_inputs_dict[input_key]["items"]]
-                input_datasets += rcp_inputs_list
-            output_key = list(rcp_outputs_dict.keys())[0]
-            rcp_outputs_list = [x["ref"] for x in rcp_outputs_dict[output_key]["items"]]
-            # Append them to the overall input list:
-            output_datasets += rcp_outputs_list
+                input_keys = list(rcp_inputs_dict.keys())
+                for input_key in input_keys:
+                    rcp_inputs_list = [x["ref"] for x in rcp_inputs_dict[input_key]["items"]]
+                    input_datasets += rcp_inputs_list
+            output_keys = list(rcp_outputs_dict.keys())
+            for output_key in output_keys:
+                rcp_outputs_list = [x["ref"] for x in rcp_outputs_dict[output_key]["items"]]
+                # Append them to the overall input list:
+                output_datasets += rcp_outputs_list
```

Comment:
same lines repeated twice, for input_keys and for output_keys : instead, let's create a method and call it twice

Reply:
Fixed in commit [f376fca95a4965dbf81268765b9fde2c29b18b87](https://github.com/gbetegon88/dataiku-contrib/commit/f376fca95a4965dbf81268765b9fde2c29b18b87)

Let me know if this should better be in the macro lib and imported in the runnable.py

Reply:
You could indeed create a utils.py in the macro lib 👍

Reply:
Done in commits:

- [34e25f63f0d57e471b30948a2e46df47a9648117](https://github.com/gbetegon88/dataiku-contrib/commit/34e25f63f0d57e471b30948a2e46df47a9648117)
- [96f6c03c22995f1fbe4187d4893d9d00617b07a9](https://github.com/gbetegon88/dataiku-contrib/commit/96f6c03c22995f1fbe4187d4893d9d00617b07a9)